nixos/modules/services/system/self-deploy.nix

```
@@ -125,13 +125,15 @@ in
   };
 
   config = lib.mkIf cfg.enable {
-    systemd.services.self-deploy = {
+    systemd.services.self-deploy = rec {
       inherit (cfg) startAt;
 
       wantedBy = [ "multi-user.target" ];
 
       requires = lib.mkIf (!(isPathType cfg.repository)) [ "network-online.target" ];
 
+      after = requires;
+
```

I believe the right thing to do here is to remove the after and requires and instead specify `serviceConfig.Type = "oneshot"`.  This should disable the service until the timer fires.

```suggestion
    systemd.services.self-deploy = {
      inherit (cfg) startAt;

      serviceConfig.Type = "oneshot";

```

---

I think we should still have `requires` and `after` in case the timer would happen around system boot or some network donwtime (if that's how network-online target even works)

---

I'll make the service `oneshot` and remove the `wantedBy` option, since it will be started once its timer fires. This should resolve the potential restart loop upon system boot.

I also think that it's fine to leave the `after` and `requires` options in place, since it won't affect normal startup, but fixes the edge case of the timer firing soon after system startup.

---

Done